### PR TITLE
Make imshow defaults the same as add_image

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description

This updates the default argument values of `napari.imshow` to be the same as `ViewerModel.add_image`. This results in more consistent and desirable, especially when using the `channel_axis` argument.

I also copied over the corresponding docstrings from `add_image`. And also fixed a few typos.

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] I manually tested `imshow` and `add_image` for consistency.
- [x] All existing tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/_tests/test_viewer_layer_parity.py [hunk 1]
```diff
@@ -6,33 +6,28 @@
...
-    # ensure both have the same parameters
-    assert set(class_parameters) == set(method_parameters)
+    # Ensure both have the same parameter names
+    assert imshow_parameters.keys() == viewer_parameters.keys()
 
-    # ensure the parameters have the same defaults
-    for name, parameter in class_parameters.items():
+    # Ensure the parameters have the same defaults
+    for name, parameter in viewer_parameters.items():
+        # data is a required for imshow, but optional for add_image
```
Comment (andy-sweet) [inline napari/_tests/test_viewer_layer_parity.py:29]: `Viewer.add_image` has the default `data=None`, though I'm unsure if that default value can ever succeed. I'm tempted to remove the default value from `add_image` to make it consistent with `imshow`, though a little wary of doing that right before the release. Any thoughts?

Comment (jni) [reply napari/_tests/test_viewer_layer_parity.py:29]: Indeed, `viewer.add_image()` fails. I'd be ok with removing it but am also nervous like you — perhaps we should leave that action until after the release?

**Review by jni — APPROVED** [review]
Awesome, thank you for taking care of this @andy-sweet! Just pulled it down and tested it and it looks great! 🎉

**Comment by andy-sweet** [comment]
Thanks for the quick review @jni! I'll likely merge this after 48 hours unless anyone objects.

**Review by kcpevey — COMMENTED** [review]
Look great. I see how I missed a few things in my implementation. Thanks @andy-sweet for cleaning this up!

File: napari/_tests/test_viewer_layer_parity.py (hunk 1 above)
Comment (andy-sweet) [reply napari/_tests/test_viewer_layer_parity.py:29]: Yeah, my plan is to remove the `None` default in `add_image`, but after the release.

**Pull request merged**